frame/bounties/src/subbounty_migration.rs

```diff
@@ -0,0 +1,166 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Migrations to version [`4.0.0`], as denoted by the changelog.
+
+use codec::{Encode, Decode, FullCodec};
+use sp_std::prelude::*;
+use frame_support::{
+	weights::Weight, Twox64Concat,
+	storage::types::{StorageMap},
+	crate_to_pallet_version,
+};
+
+use sp_runtime::RuntimeDebug;
+
+/// An index of a bounty. Just a `u32`.
+pub type BountyIndex = u32;
+
+/// Trait to implement to give information about types used for migration
+pub trait SubBountyMigration {
+
+	type AccountId: 'static + FullCodec;
+
+	type BlockNumber: 'static + FullCodec;
+
+	type Balance: 'static + FullCodec + Copy;
+}
+
+/// The status of a bounty proposal.
```

Comment:
Documentation should state that these are historical storage types and we only duplciate them to make the migrations reusable.

Reply:
Documentation update done. :+1: